File: src/main/java/com/mssoftech/dbflute/bsentity/dbmeta/MemberAddressDbm.java
package com.mssoftech.dbflute.bsentity.dbmeta;

import java.util.List;
import java.util.Map;

import org.dbflute.Entity;
import org.dbflute.optional.OptionalEntity;
import org.dbflute.dbmeta.AbstractDBMeta;
import org.dbflute.dbmeta.info.*;
import org.dbflute.dbmeta.name.*;
import org.dbflute.dbmeta.property.PropertyGateway;
import org.dbflute.dbway.DBDef;
import com.mssoftech.dbflute.allcommon.*;
import com.mssoftech.dbflute.exentity.*;

/**
 * The DB meta of member_address. (Singleton)
 * @author DBFlute(AutoGenerator)
 */
public class MemberAddressDbm extends AbstractDBMeta {

    // ===================================================================================
    //                                                                           Singleton
    //                                                                           =========
    private static final MemberAddressDbm _instance = new MemberAddressDbm();
    private MemberAddressDbm() {}
    public static MemberAddressDbm getInstance() { return _instance; }

    // ===================================================================================
    //                                                                       Current DBDef
    //                                                                       =============
    public String getProjectName() { return DBCurrent.getInstance().projectName(); }
    public String getProjectPrefix() { return DBCurrent.getInstance().projectPrefix(); }
    public String getGenerationGapBasePrefix() { return DBCurrent.getInstance().generationGapBasePrefix(); }
    public DBDef getCurrentDBDef() { return DBCurrent.getInstance().currentDBDef(); }

    // ===================================================================================
    //                                                                    Property Gateway
    //                                                                    ================
    // -----------------------------------------------------
    //                                       Column Property
    //                                       ---------------
    protected final Map<String, PropertyGateway> _epgMap = newHashMap();
    { xsetupEpg(); }
    protected void xsetupEpg() {
        setupEpg(_epgMap, et -> ((MemberAddress)et).getMemberAddressId(), (et, vl) -> ((MemberAddress)et).setMemberAddressId(cti(vl)), "memberAddressId");
        setupEpg(_epgMap, et -> ((MemberAddress)et).getMemberId(), (et, vl) -> ((MemberAddress)et).setMemberId(cti(vl)), "memberId");
        setupEpg(_epgMap, et -> ((MemberAddress)et).getValidBeginDate(), (et, vl) -> ((MemberAddress)et).setValidBeginDate(ctld(vl)), "validBeginDate");
        setupEpg(_epgMap, et -> ((MemberAddress)et).getValidEndDate(), (et, vl) -> ((MemberAddress)et).setValidEndDate(ctld(vl)), "validEndDate");
        setupEpg(_epgMap, et -> ((MemberAddress)et).getAddress(), (et, vl) -> ((MemberAddress)et).setAddress((String)vl), "address");
        setupEpg(_epgMap, et -> ((MemberAddress)et).getRegionId(), (et, vl) -> ((MemberAddress)et).setRegionId(cti(vl)), "regionId");
        setupEpg(_epgMap, et -> ((MemberAddress)et).getRegisterDatetime(), (et, vl) -> ((MemberAddress)et).setRegisterDatetime(ctldt(vl)), "registerDatetime");
        setupEpg(_epgMap, et -> ((MemberAddress)et).getRegisterProcess(), (et, vl) -> ((MemberAddress)et).setRegisterProcess((String)vl), "registerProcess");
        setupEpg(_epgMap, et -> ((MemberAddress)et).getRegisterUser(), (et, vl) -> ((MemberAddress)et).setRegisterUser((String)vl), "registerUser");
        setupEpg(_epgMap, et -> ((MemberAddress)et).getUpdateDatetime(), (et, vl) -> ((MemberAddress)et).setUpdateDatetime(ctldt(vl)), "updateDatetime");
        setupEpg(_epgMap, et -> ((MemberAddress)et).getUpdateProcess(), (et, vl) -> ((MemberAddress)et).setUpdateProcess((String)vl), "updateProcess");
        setupEpg(_epgMap, et -> ((MemberAddress)et).getUpdateUser(), (et, vl) -> ((MemberAddress)et).setUpdateUser((String)vl), "updateUser");
        setupEpg(_epgMap, et -> ((MemberAddress)et).getVersionNo(), (et, vl) -> ((MemberAddress)et).setVersionNo(ctl(vl)), "versionNo");
    }
    public PropertyGateway findPropertyGateway(String prop)
    { return doFindEpg(_epgMap, prop); }

    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------
    protected final Map<String, PropertyGateway> _efpgMap = newHashMap();
    { xsetupEfpg(); }
    @SuppressWarnings("unchecked")
    protected void xsetupEfpg() {
        setupEfpg(_efpgMap, et -> ((MemberAddress)et).getMember(), (et, vl) -> ((MemberAddress)et).setMember((OptionalEntity<Member>)vl), "member");
        setupEfpg(_efpgMap, et -> ((MemberAddress)et).getRegion(), (et, vl) -> ((MemberAddress)et).setRegion((OptionalEntity<Region>)vl), "region");
    }
    public PropertyGateway findForeignPropertyGateway(String prop)
    { return doFindEfpg(_efpgMap, prop); }

    // ===================================================================================
    //                                                                          Table Info
    //                                                                          ==========
    protected final String _tableDbName = "member_address";
    protected final String _tableDispName = "member_address";
    protected final String _tablePropertyName = "memberAddress";
    protected final TableSqlName _tableSqlName = new TableSqlName("member_address", _tableDbName);
    { _tableSqlName.xacceptFilter(DBFluteConfig.getInstance().getTableSqlNameFilter()); }
    public String getTableDbName() { return _tableDbName; }
    public String getTableDispName() { return _tableDispName; }
    public String getTablePropertyName() { return _tablePropertyName; }
    public TableSqlName getTableSqlName() { return _tableSqlName; }

    // ===================================================================================
    //                                                                         Column Info
    //                                                                         ===========
    protected final ColumnInfo _columnMemberAddressId = cci("member_address_id", "member_address_id", null, null, Integer.class, "memberAddressId", null, true, true, true, "serial", 10, 0, "nextval('member_address_member_address_id_seq'::regclass)", false, null, null, null, null, null, false);
//"int64"
    protected final ColumnInfo _columnMemberId = cci("member_id", "member_id", null, null, Integer.class, "memberId", null, false, false, true, "int4", 10, 0, null, false, null, null, "member", null, null, false);
//"int64"
    protected final ColumnInfo _columnValidBeginDate = cci("valid_begin_date", "valid_begin_date", null, null, java.time.LocalDate.class, "validBeginDate", null, false, false, true, "date", 13, 0, null, false, null, null, null, null, null, false);
//"df.Date"
    protected final ColumnInfo _columnValidEndDate = cci("valid_end_date", "valid_end_date", null, null, java.time.LocalDate.class, "validEndDate", null, false, false, true, "date", 13, 0, null, false, null, null, null, null, null, false);
//"df.Date"
    protected final ColumnInfo _columnAddress = cci("address", "address", null, null, String.class, "address", null, false, false, true, "varchar", 200, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnRegionId = cci("region_id", "region_id", null, null, Integer.class, "regionId", null, false, false, true, "int4", 10, 0, null, false, null, null, "region", null, null, false);
//"int64"
    protected final ColumnInfo _columnRegisterDatetime = cci("register_datetime", "register_datetime", null, null, java.time.LocalDateTime.class, "registerDatetime", null, false, false, true, "timestamp", 26, 3, null, false, null, null, null, null, null, false);
//"df.Timestamp"
    protected final ColumnInfo _columnRegisterProcess = cci("register_process", "register_process", null, null, String.class, "registerProcess", null, false, false, true, "varchar", 200, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnRegisterUser = cci("register_user", "register_user", null, null, String.class, "registerUser", null, false, false, true, "varchar", 200, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnUpdateDatetime = cci("update_datetime", "update_datetime", null, null, java.time.LocalDateTime.class, "updateDatetime", null, false, false, true, "timestamp", 26, 3, null, false, null, null, null, null, null, false);
//"df.Timestamp"
    protected final ColumnInfo _columnUpdateProcess = cci("update_process", "update_process", null, null, String.class, "updateProcess", null, false, false, true, "varchar", 200, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnUpdateUser = cci("update_user", "update_user", null, null, String.class, "updateUser", null, false, false, true, "varchar", 200, 0, null, false, null, null, null, null, null, false);
//"string"
    protected final ColumnInfo _columnVersionNo = cci("version_no", "version_no", null, null, Long.class, "versionNo", null, false, false, true, "int8", 19, 0, null, false, OptimisticLockType.VERSION_NO, null, null, null, null, false);
//"int64"

    /**
     * member_address_id: {PK, ID, NotNull, serial(10)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnMemberAddressId() { return _columnMemberAddressId; }
    /**
     * member_id: {UQ+, NotNull, int4(10), FK to member}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnMemberId() { return _columnMemberId; }
    /**
     * valid_begin_date: {+UQ, NotNull, date(13)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnValidBeginDate() { return _columnValidBeginDate; }
    /**
     * valid_end_date: {NotNull, date(13)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnValidEndDate() { return _columnValidEndDate; }
    /**
     * address: {NotNull, varchar(200)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnAddress() { return _columnAddress; }
    /**
     * region_id: {NotNull, int4(10), FK to region}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnRegionId() { return _columnRegionId; }
    /**
     * register_datetime: {NotNull, timestamp(26, 3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnRegisterDatetime() { return _columnRegisterDatetime; }
    /**
     * register_process: {NotNull, varchar(200)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnRegisterProcess() { return _columnRegisterProcess; }
    /**
     * register_user: {NotNull, varchar(200)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnRegisterUser() { return _columnRegisterUser; }
    /**
     * update_datetime: {NotNull, timestamp(26, 3)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnUpdateDatetime() { return _columnUpdateDatetime; }
    /**
     * update_process: {NotNull, varchar(200)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnUpdateProcess() { return _columnUpdateProcess; }
    /**
     * update_user: {NotNull, varchar(200)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnUpdateUser() { return _columnUpdateUser; }
    /**
     * version_no: {NotNull, int8(19)}
     * @return The information object of specified column. (NotNull)
     */
    public ColumnInfo columnVersionNo() { return _columnVersionNo; }

    protected List<ColumnInfo> ccil() {
        List<ColumnInfo> ls = newArrayList();
        ls.add(columnMemberAddressId());
        ls.add(columnMemberId());
        ls.add(columnValidBeginDate());
        ls.add(columnValidEndDate());
        ls.add(columnAddress());
        ls.add(columnRegionId());
        ls.add(columnRegisterDatetime());
        ls.add(columnRegisterProcess());
        ls.add(columnRegisterUser());
        ls.add(columnUpdateDatetime());
        ls.add(columnUpdateProcess());
        ls.add(columnUpdateUser());
        ls.add(columnVersionNo());
        return ls;
    }

    { initializeInformationResource(); }

    // ===================================================================================
    //                                                                         Unique Info
    //                                                                         ===========
    // -----------------------------------------------------
    //                                       Primary Element
    //                                       ---------------
    protected UniqueInfo cpui() { return hpcpui(columnMemberAddressId()); }
    public boolean hasPrimaryKey() { return true; }
    public boolean hasCompoundPrimaryKey() { return false; }

    // -----------------------------------------------------
    //                                        Unique Element
    //                                        --------------
    public UniqueInfo uniqueOf() {
        List<ColumnInfo> ls = newArrayListSized(4);
        ls.add(columnMemberId());
        ls.add(columnValidBeginDate());
        return hpcui(ls);
    }

    // ===================================================================================
    //                                                                       Relation Info
    //                                                                       =============
    // cannot cache because it uses related DB meta instance while booting
    // (instead, cached by super's collection)
    // -----------------------------------------------------
    //                                      Foreign Property
    //                                      ----------------
    /**
     * member by my member_id, named 'member'.
     * @return The information object of foreign property. (NotNull)
     */
    public ForeignInfo foreignMember() {
        Map<ColumnInfo, ColumnInfo> mp = newLinkedHashMap(columnMemberId(), MemberDbm.getInstance().columnMemberId());
        return cfi("fk_member_address_member", "member", this, MemberDbm.getInstance(), mp, 0, org.dbflute.optional.OptionalEntity.class, false, false, false, false, null, null, false, "memberAddressList", false);
    }
    /**
     * region by my region_id, named 'region'.
     * @return The information object of foreign property. (NotNull)
     */
    public ForeignInfo foreignRegion() {
        Map<ColumnInfo, ColumnInfo> mp = newLinkedHashMap(columnRegionId(), RegionDbm.getInstance().columnRegionId());
        return cfi("fk_member_address_region", "region", this, RegionDbm.getInstance(), mp, 1, org.dbflute.optional.OptionalEntity.class, false, false, false, false, null, null, false, "memberAddressList", false);
    }

    // -----------------------------------------------------
    //                                     Referrer Property
    //                                     -----------------

    // ===================================================================================
    //                                                                        Various Info
    //                                                                        ============
    public boolean hasSequence() { return true; }
    public String getSequenceName() { return "member_address_member_address_id_seq"; }
    public Integer getSequenceIncrementSize() { return 1; }
    public Integer getSequenceCacheSize() { return null; }
    public boolean hasVersionNo() { return true; }
    public ColumnInfo getVersionNoColumnInfo() { return _columnVersionNo; }

    // ===================================================================================
    //                                                                           Type Name
    //                                                                           =========
    public String getEntityTypeName() { return "com.mssoftech.dbflute.exentity.MemberAddress"; }
    public String getConditionBeanTypeName() { return "com.mssoftech.dbflute.cbean.MemberAddressCB"; }
    public String getBehaviorTypeName() { return "com.mssoftech.dbflute.exbhv.MemberAddressBhv"; }

    // ===================================================================================
    //                                                                         Object Type
    //                                                                         ===========
    public Class<MemberAddress> getEntityType() { return MemberAddress.class; }

    // ===================================================================================
    //                                                                     Object Instance
    //                                                                     ===============
    public MemberAddress newEntity() { return new MemberAddress(); }

    // ===================================================================================
    //                                                                   Map Communication
    //                                                                   =================
    public void acceptPrimaryKeyMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptPrimaryKeyMap((MemberAddress)et, mp); }
    public void acceptAllColumnMap(Entity et, Map<String, ? extends Object> mp)
    { doAcceptAllColumnMap((MemberAddress)et, mp); }
    public Map<String, Object> extractPrimaryKeyMap(Entity et) { return doExtractPrimaryKeyMap(et); }
    public Map<String, Object> extractAllColumnMap(Entity et) { return doExtractAllColumnMap(et); }
}

File: src/dbflute/adf/meta/memberAddressdbm.go
package meta

import (
	"github.com/mikeshimura/dbflute/df"
)

type MemberAddressDbm_T struct {
	df.BaseDBMeta
	ColumnMemberAddressId *df.ColumnInfo
	ColumnMemberId *df.ColumnInfo
	ColumnValidBeginDate *df.ColumnInfo
	ColumnValidEndDate *df.ColumnInfo
	ColumnAddress *df.ColumnInfo
	ColumnRegionId *df.ColumnInfo
	ColumnRegisterDatetime *df.ColumnInfo
	ColumnRegisterProcess *df.ColumnInfo
	ColumnRegisterUser *df.ColumnInfo
	ColumnUpdateDatetime *df.ColumnInfo
	ColumnUpdateProcess *df.ColumnInfo
	ColumnUpdateUser *df.ColumnInfo
	ColumnVersionNo *df.ColumnInfo
}

func (b *MemberAddressDbm_T) GetProjectName() string {
	return df.DBCurrent_I.ProjectName
}

func (b *MemberAddressDbm_T) GetDbCurrent() *df.DBCurrent {
	return df.DBCurrent_I
}

var MemberAddressDbm *MemberAddressDbm_T

func Create_MemberAddressDbm() {
	MemberAddressDbm = new(MemberAddressDbm_T)
	MemberAddressDbm.TableDbName = "member_address"
	MemberAddressDbm.TableDispName = "member_address"
	MemberAddressDbm.TablePropertyName = "memberAddress"
	MemberAddressDbm.TableSqlName = new(df.TableSqlName)
	MemberAddressDbm.TableSqlName.TableSqlName = "member_address"
	MemberAddressDbm.TableSqlName.CorrespondingDbName = MemberAddressDbm.TableDbName

	var memberAddress df.DBMeta
	memberAddress = MemberAddressDbm
	MemberAddressDbm.DBMeta=&memberAddress
	memberAddressIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo member_address_id
	memberAddressIdSqlName.ColumnSqlName = "member_address_id"
	memberAddressIdSqlName.IrregularChar = false
	MemberAddressDbm.ColumnMemberAddressId = df.CCI(&memberAddress, "member_address_id", memberAddressIdSqlName, "", "", "Integer.class", "memberAddressId", "", true, true,true, "serial", 10, 0, "nextval('member_address_member_address_id_seq'::regclass)",false,"","", "","","",false,"int64")
	memberIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo member_id
	memberIdSqlName.ColumnSqlName = "member_id"
	memberIdSqlName.IrregularChar = false
	MemberAddressDbm.ColumnMemberId = df.CCI(&memberAddress, "member_id", memberIdSqlName, "", "", "Integer.class", "memberId", "", false, false,true, "int4", 10, 0, "",false,"","", "member","","",false,"int64")
	validBeginDateSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo valid_begin_date
	validBeginDateSqlName.ColumnSqlName = "valid_begin_date"
	validBeginDateSqlName.IrregularChar = false
	MemberAddressDbm.ColumnValidBeginDate = df.CCI(&memberAddress, "valid_begin_date", validBeginDateSqlName, "", "", "java.time.LocalDate.class", "validBeginDate", "", false, false,true, "date", 13, 0, "",false,"","", "","","",false,"df.Date")
	validEndDateSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo valid_end_date
	validEndDateSqlName.ColumnSqlName = "valid_end_date"
	validEndDateSqlName.IrregularChar = false
	MemberAddressDbm.ColumnValidEndDate = df.CCI(&memberAddress, "valid_end_date", validEndDateSqlName, "", "", "java.time.LocalDate.class", "validEndDate", "", false, false,true, "date", 13, 0, "",false,"","", "","","",false,"df.Date")
	addressSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo address
	addressSqlName.ColumnSqlName = "address"
	addressSqlName.IrregularChar = false
	MemberAddressDbm.ColumnAddress = df.CCI(&memberAddress, "address", addressSqlName, "", "", "String.class", "address", "", false, false,true, "varchar", 200, 0, "",false,"","", "","","",false,"string")
	regionIdSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo region_id
	regionIdSqlName.ColumnSqlName = "region_id"
	regionIdSqlName.IrregularChar = false
	MemberAddressDbm.ColumnRegionId = df.CCI(&memberAddress, "region_id", regionIdSqlName, "", "", "Integer.class", "regionId", "", false, false,true, "int4", 10, 0, "",false,"","", "region","","",false,"int64")
	registerDatetimeSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo register_datetime
	registerDatetimeSqlName.ColumnSqlName = "register_datetime"
	registerDatetimeSqlName.IrregularChar = false
	MemberAddressDbm.ColumnRegisterDatetime = df.CCI(&memberAddress, "register_datetime", registerDatetimeSqlName, "", "", "java.time.LocalDateTime.class", "registerDatetime", "", false, false,true, "timestamp", 26, 3, "",false,"","", "","","",false,"df.Timestamp")
	registerProcessSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo register_process
	registerProcessSqlName.ColumnSqlName = "register_process"
	registerProcessSqlName.IrregularChar = false
	MemberAddressDbm.ColumnRegisterProcess = df.CCI(&memberAddress, "register_process", registerProcessSqlName, "", "", "String.class", "registerProcess", "", false, false,true, "varchar", 200, 0, "",false,"","", "","","",false,"string")
	registerUserSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo register_user
	registerUserSqlName.ColumnSqlName = "register_user"
	registerUserSqlName.IrregularChar = false
	MemberAddressDbm.ColumnRegisterUser = df.CCI(&memberAddress, "register_user", registerUserSqlName, "", "", "String.class", "registerUser", "", false, false,true, "varchar", 200, 0, "",false,"","", "","","",false,"string")
	updateDatetimeSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo update_datetime
	updateDatetimeSqlName.ColumnSqlName = "update_datetime"
	updateDatetimeSqlName.IrregularChar = false
	MemberAddressDbm.ColumnUpdateDatetime = df.CCI(&memberAddress, "update_datetime", updateDatetimeSqlName, "", "", "java.time.LocalDateTime.class", "updateDatetime", "", false, false,true, "timestamp", 26, 3, "",false,"","", "","","",false,"df.Timestamp")
	updateProcessSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo update_process
	updateProcessSqlName.ColumnSqlName = "update_process"
	updateProcessSqlName.IrregularChar = false
	MemberAddressDbm.ColumnUpdateProcess = df.CCI(&memberAddress, "update_process", updateProcessSqlName, "", "", "String.class", "updateProcess", "", false, false,true, "varchar", 200, 0, "",false,"","", "","","",false,"string")
	updateUserSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo update_user
	updateUserSqlName.ColumnSqlName = "update_user"
	updateUserSqlName.IrregularChar = false
	MemberAddressDbm.ColumnUpdateUser = df.CCI(&memberAddress, "update_user", updateUserSqlName, "", "", "String.class", "updateUser", "", false, false,true, "varchar", 200, 0, "",false,"","", "","","",false,"string")
	versionNoSqlName := new(df.ColumnSqlName)
	//colsqlname dayoo version_no
	versionNoSqlName.ColumnSqlName = "version_no"
	versionNoSqlName.IrregularChar = false
	MemberAddressDbm.ColumnVersionNo = df.CCI(&memberAddress, "version_no", versionNoSqlName, "", "", "Long.class", "versionNo", "", false, false,true, "int8", 19, 0, "",false,"OptimisticLockType.VERSION_NO","", "","","",false,"int64")

	MemberAddressDbm.ColumnInfoList = new(df.List)
	MemberAddressDbm.ColumnInfoList.Add(MemberAddressDbm.ColumnMemberAddressId)
	MemberAddressDbm.ColumnInfoList.Add(MemberAddressDbm.ColumnMemberId)
	MemberAddressDbm.ColumnInfoList.Add(MemberAddressDbm.ColumnValidBeginDate)
	MemberAddressDbm.ColumnInfoList.Add(MemberAddressDbm.ColumnValidEndDate)
	MemberAddressDbm.ColumnInfoList.Add(MemberAddressDbm.ColumnAddress)
	MemberAddressDbm.ColumnInfoList.Add(MemberAddressDbm.ColumnRegionId)
	MemberAddressDbm.ColumnInfoList.Add(MemberAddressDbm.ColumnRegisterDatetime)
	MemberAddressDbm.ColumnInfoList.Add(MemberAddressDbm.ColumnRegisterProcess)
	MemberAddressDbm.ColumnInfoList.Add(MemberAddressDbm.ColumnRegisterUser)
	MemberAddressDbm.ColumnInfoList.Add(MemberAddressDbm.ColumnUpdateDatetime)
	MemberAddressDbm.ColumnInfoList.Add(MemberAddressDbm.ColumnUpdateProcess)
	MemberAddressDbm.ColumnInfoList.Add(MemberAddressDbm.ColumnUpdateUser)
	MemberAddressDbm.ColumnInfoList.Add(MemberAddressDbm.ColumnVersionNo)


	MemberAddressDbm.ColumnInfoMap=make(map[string]int)
	MemberAddressDbm.ColumnInfoMap["memberAddressId"]=0
		MemberAddressDbm.ColumnInfoMap["memberId"]=1
		MemberAddressDbm.ColumnInfoMap["validBeginDate"]=2
		MemberAddressDbm.ColumnInfoMap["validEndDate"]=3
		MemberAddressDbm.ColumnInfoMap["address"]=4
		MemberAddressDbm.ColumnInfoMap["regionId"]=5
		MemberAddressDbm.ColumnInfoMap["registerDatetime"]=6
		MemberAddressDbm.ColumnInfoMap["registerProcess"]=7
		MemberAddressDbm.ColumnInfoMap["registerUser"]=8
		MemberAddressDbm.ColumnInfoMap["updateDatetime"]=9
		MemberAddressDbm.ColumnInfoMap["updateProcess"]=10
		MemberAddressDbm.ColumnInfoMap["updateUser"]=11
		MemberAddressDbm.ColumnInfoMap["versionNo"]=12
	    MemberAddressDbm.PrimaryKey = true
    MemberAddressDbm.CompoundPrimaryKey = false
	ui := new(df.UniqueInfo)
	ui.DbMeta = &memberAddress
	ui.Primary = true
	ui.UniqueColumnList = new(df.List)
	ui.UniqueColumnList.Add(MemberAddressDbm.ColumnMemberAddressId)

	MemberAddressDbm.PrimaryInfo = new(df.PrimaryInfo)
	MemberAddressDbm.PrimaryInfo.UniqueInfo = ui
	
	MemberAddressDbm.SequenceFlag = true
	MemberAddressDbm.SequenceName = "member_address_member_address_id_seq"
	MemberAddressDbm.SequenceIncrementSize = 1
	MemberAddressDbm.SequenceCacheSize = 0
	MemberAddressDbm.VersionNoFlag = true
	MemberAddressDbm.VersionNoColumnInfo = MemberAddressDbm.ColumnVersionNo
	
	var memberAddressMeta df.DBMeta = MemberAddressDbm
	df.DBMetaInstanceHandler_I.TableDbNameInstanceMap["MemberAddress"] = &memberAddressMeta
}
